src/groupByKey.ts

```diff
@@ -2,6 +2,22 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(object: T[], key: keyof T): GroupsMap<T> {
+  const finalObject: GroupsMap<T> = {};
+
+  object.forEach((el) => {
+    const keyValue = el[key];
+
+    if (keyValue !== undefined && keyValue !== null) {
+      const keyString = String(keyValue);
+
+      if (!finalObject.hasOwnProperty(keyString)) {
```

Comment:
Instead of using `hasOwnProperty`, consider using `Object.prototype.hasOwnProperty.call(finalObject, keyString)` for checking the existence of a key. This is to prevent issues that might arise if the `finalObject` has a property with the same name as a method from `Object.prototype` (e.g., `hasOwnProperty`).